### PR TITLE
Feature full authentication send

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/concent/issues/209

### CODE REVIEW

File: concent_api/core/views.py [hunk 1]
```diff
@@ -3,19 +3,51 @@
...
 from utils.api_view                 import api_view
 from utils.helpers                  import decode_client_public_key
+from utils.shortcuts                import load_without_public_key
 from .models                        import PendingResponse
 
 
 @api_view
 @require_POST
 def send(request, client_message):
+    if len(request.body) == 0:
```
Comment (dybi) [inline concent_api/core/views.py]: please extract it to a functi0on `validate_request` or sth similar and move it to `validations` module

File: concent_api/core/message_handlers.py [hunk 2]
```diff
@@ -66,32 +68,35 @@ def handle_send_force_report_computed_task(request, client_message):
     subtask = store_subtask(
         task_id              = client_message.report_computed_task.task_to_compute.compute_task_def['task_id'],
         subtask_id           = client_message.report_computed_task.task_to_compute.compute_task_def['subtask_id'],
-        provider_public_key  = client_public_key,
-        requestor_public_key = other_party_public_key,
+        provider_public_key  = client_message.report_computed_task.task_to_compute.provider_public_key,
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: Why ? It is used only once, and it makes my much more sense to show from where it is coming from - `TaskToCompute` in this case. I am not a fan of creating extra variables if there is no need to and in this case there is not. Also, why only this one ? There are lot of others values get this way in code above and below.

File: concent_api/core/message_handlers.py [hunk 3]
```diff
@@ -66,32 +68,35 @@ def handle_send_force_report_computed_task(request, client_message):
...
+        requestor_public_key = client_message.report_computed_task.task_to_compute.requestor_public_key,
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: Same here.

File: concent_api/core/message_handlers.py [hunk 4]
```diff
@@ -66,32 +68,35 @@ def handle_send_force_report_computed_task(request, client_message):
...
     logging.log_message_added_to_queue(
         client_message,
-        client_public_key,
+        client_message.report_computed_task.task_to_compute.provider_public_key,
     )
     return HttpResponse("", status = 202)
 
 
-def handle_send_ack_report_computed_task(request, client_message):
+def handle_send_ack_report_computed_task(_request, client_message):
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: It is used in several other `send` handlers. I wanted those functions to keep similar signature just to make the code more readable. And it also would be needed if we want to refactor them somehow.

File: concent_api/core/message_handlers.py [hunk 5]
```diff
@@ -66,32 +68,35 @@ def handle_send_force_report_computed_task(request, client_message):
...
     current_time      = get_current_utc_timestamp()
-    client_public_key = decode_client_public_key(request)
     validate_golem_message_task_to_compute(client_message.task_to_compute)
     validate_id_value(client_message.task_to_compute.compute_task_def['task_id'], 'task_id')
     validate_id_value(client_message.task_to_compute.compute_task_def['subtask_id'], 'subtask_id')
+    validate_golem_message_signed_with_key(
+        client_message.task_to_compute,
+        client_message.task_to_compute.provider_public_key,
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: Same here.

File: concent_api/core/message_handlers.py [hunk 6]
```diff
@@ -133,24 +143,27 @@ def handle_send_ack_report_computed_task(request, client_message):
...
             client_message,
-            request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'],
+            client_message.task_to_compute.requestor_public_key,
             client_message.task_to_compute.compute_task_def['deadline'] + settings.CONCENT_MESSAGING_TIME,
         )
         raise Http400("Time to acknowledge this task is already over.")
 
 
-def handle_send_reject_report_computed_task(request, client_message):
+def handle_send_reject_report_computed_task(_request, client_message):
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: Please check my answer above.

File: concent_api/core/message_handlers.py [hunk 7]
```diff
@@ -217,27 +235,29 @@ def handle_send_reject_report_computed_task(request, client_message):
...
             client_message,
-            request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'],
+            client_message.task_to_compute.requestor_public_key,
             deserialized_message.compute_task_def['deadline'] + settings.CONCENT_MESSAGING_TIME,
         )
         raise Http400("Time to acknowledge this task is already over.")
 
 
-def handle_send_force_get_task_result(request, client_message: message.concents.ForceGetTaskResult) -> message.concents:
+def handle_send_force_get_task_result(_request, client_message: message.concents.ForceGetTaskResult) -> message.concents:
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: Please check my answer above.

File: concent_api/core/message_handlers.py [hunk 8]
```diff
@@ -348,47 +370,46 @@ def handle_send_force_subtask_results(request, client_message: message.concents.
...
         logging.log_message_added_to_queue(
             client_message,
-            client_public_key,
+            client_message.ack_report_computed_task.task_to_compute.provider_public_key,
         )
         return HttpResponse("", status = 202)
 
 
-def handle_send_force_subtask_results_response(request, client_message):
+def handle_send_force_subtask_results_response(_request, client_message):
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: Please check my answer above.

File: concent_api/core/views.py (hunk 1 above)
Comment (rwrzesien) [reply concent_api/core/views.py]: This piece of code was removed from here and will be back in `api_view` decorator.

File: concent_api/utils/tests/test_decorators.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+from utils.testing_helpers          import generate_ecc_key_pair
+
+
+# (CONCENT_PRIVATE_KEY,   CONCENT_PUBLIC_KEY)   = generate_ecc_key_pair()
+
+# @override_settings(
+#     CONCENT_PRIVATE_KEY = CONCENT_PRIVATE_KEY,
+#     CONCENT_PUBLIC_KEY  = CONCENT_PUBLIC_KEY,
+# )
+# class DecoratorsTestCase(ConcentIntegrationTestCase):
```
Comment (Jakub89) [inline concent_api/utils/tests/test_decorators.py]: What's the point of that whole file?

File: concent_api/core/tests/utils.py [hunk 10]
```diff
@@ -625,3 +649,24 @@ def _add_signature_to_message(self, golem_message, priv_key):  # pylint: disable
...
+
+    def _create_requestor_auth_message(self):
+        return self._create_client_auth_message(self.REQUESTOR_PRIVATE_KEY, self.REQUESTOR_PUBLIC_KEY)
+
+    def _create_diff_requestor_auth_message(self):
+        return self._create_client_auth_message(self.DIFFERENT_REQUESTOR_PRIVATE_KEY, self.DIFFERENT_REQUESTOR_PUBLIC_KEY)
+
+    def _create_test_ping_message(self):
+        self.ping_message = message.Ping()
+        return self.ping_message
```
Comment (Jakub89) [inline concent_api/core/tests/utils.py]: "C:670, 0: Final newline missing (missing-final-newline)"
There is a bunch of pylint errors that need to be resolved.

File: concent_api/utils/tests/test_decorators.py (hunk 9 above)
Comment (rwrzesien) [reply concent_api/utils/tests/test_decorators.py]: Yes, it is rebased on `feature-add-auth-decorator` branch which is still in progress, and this file are tests which has to be finished there. I have changed PR base branch so you will see only changes from this one. Please note this is kind-of-a-wip branch but it is not expecting any new code apart from the final rebase on `feature-add-auth-decorator`.

File: concent_api/core/tests/utils.py (hunk 10 above)
Comment (rwrzesien) [reply concent_api/core/tests/utils.py]: Please check my comment above.

**Pull request closed without merging**